### PR TITLE
babe: introduce a request-answering mechanic

### PR DESCRIPTION
**Pull request by rphmeier**
For now, the only request type it answers is requesting the epoch that a child block will have.

### CODE REVIEW

**Comment by rphmeier** [comment]
I'm not totally sure I need this, given #7789, but maybe it's useful? Cheaper than going to runtime for sure.

File: client/consensus/babe/src/lib.rs [hunk 1]
```diff
@@ -446,32 +449,104 @@ pub fn start_babe<B, C, SC, E, I, SO, CAW, BS, Error>(BabeParams {
...
+	/// The parent block is identified by its hash and number.
+	EpochForChild(
+		B::Hash,
+		NumberFor<B>,
+		SlotNumber,
+		oneshot::Sender<Result<sp_consensus_babe::Epoch, Error<B>>>,
+	),
+
+	#[doc(hidden)]
+	__Nonexhaustive,
```
Comment (bkchr) [inline client/consensus/babe/src/lib.rs:537]: Why not https://doc.rust-lang.org/reference/attributes/type_system.html?

**Review by andresilva — APPROVED** [review]
I think we might be able to use this for BABE's RPC module, rather than exposing `SharedEpochChanges`.

File: client/consensus/babe/src/lib.rs (hunk 1 above)
Comment (rphmeier) [reply client/consensus/babe/src/lib.rs:537]: Me too :)

File: client/consensus/babe/src/lib.rs [hunk 2]
```diff
@@ -486,6 +556,11 @@ impl<B: BlockT> BabeWorker<B> {
 		self.slot_notification_sinks.lock().push(sink);
 		stream
 	}
+
+	/// Get a handle to the worker.
+	pub fn handle(&self) -> BabeWorkerHandle<B> {
+		self.handle.clone()
```
Comment (seunlanlege) [inline client/consensus/babe/src/lib.rs:562]: This handle is for sending `BabeRequest` right? but looks like you don't expose the inner `Sender` from `BabeWorkerHandle`, unless this is still a WIP.

**Comment by gnunicorn** [comment]
any progress on this?

**Comment by rphmeier** [comment]
@gnunicorn pretty sure it's ready

**Review by bkchr — APPROVED** [review]

**Pull request merged**